src/convertToObject.js

```diff
@@ -6,7 +6,30 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+
+  // Крок 1: Очистка вхідного рядка
+  const cleanedString = sourceString
+    .replace(/[\n\r]/g, '') // Видаляємо символи нового рядка
+    .replace(/\s*;\s*/g, ';') // Видаляємо зайві пробіли навколо крапок з комою
+    .replace(/;\s*$/, ''); // Видаляємо останню крапку з комою
+
+  // Крок 2: Розділяємо рядок на окремі декларації
+  const declarations = cleanedString.split(';');
+
+  // Крок 3: Обробка кожної декларації
+  declarations.forEach((declaration) => {
+    const [property, ...valueParts] = declaration.split(':');
+
+    if (property) {
+      const value = valueParts.join(':').trim();
+
+      // Повертаємо пробіли на місце після розбиття на рядки
+      result[property.trim()] = value.replace(/,\s*/g, ',\n          ');
+    }
+  });
```

Comment:
Avoid using loops in this task, use iteration methods instead. Consider using `reduce` to build your result object, which aligns with the task requirement to avoid modifying an existing array/object and instead return a new one.